pluto/src/synnax/Provider.tsx

```diff
@@ -88,20 +83,50 @@ export const Provider = Aether.wrap<ProviderProps>(
           clusterKey: "",
           status: "connecting",
           message: "Connecting...",
+          clientServerCompatible: false,
+          clientVersion: c.clientVersion,
         },
       });
 
       const connectivity = await c.connectivity.check();
 
-      setState({
-        synnax: c,
-        state: connectivity,
-      });
+      setState({ synnax: c, state: connectivity });
       add({
         variant: CONNECTION_STATE_VARIANT[connectivity.status],
         message: connectivity.message ?? connectivity.status.toUpperCase(),
       });
 
+      if (connectivity.status === "connected" && !connectivity.clientServerCompatible) {
+        const oldServer =
+          connectivity.nodeVersion == null ||
+          migrate.semVerOlder(connectivity.nodeVersion, connectivity.clientVersion);
+
+        let description: string;
+        if (!oldServer)
+          description = `
+        Cluster version ${connectivity.nodeVersion} is newer than client version ${connectivity.clientVersion}.
+        Compatibility issues may arise. 
```

Comment:
Personally I think we should put long strings on one line, but I don't know if there is a convention followed in the rest of the codebase.